tests/test_auth.py

```diff
@@ -152,14 +153,42 @@ def test_get_username_runtime_error_suppressed(
     entered_username, keyring_no_backends_get_credential, caplog, config
 ):
     assert auth.Resolver(config, auth.CredentialInput()).username == "entered user"
-    assert caplog.messages == ["fail!"]
+    assert re.search(
+        r"Error from keyring.+Traceback.+RuntimeError: fail!", caplog.text, re.DOTALL
+    )
 
 
 def test_get_password_runtime_error_suppressed(
     entered_password, keyring_no_backends, caplog, config
 ):
     assert auth.Resolver(config, auth.CredentialInput("user")).password == "entered pw"
-    assert caplog.messages == ["fail!"]
+    assert re.search(
+        r"Error from keyring.+Traceback.+RuntimeError: fail!", caplog.text, re.DOTALL
+    )
+
+
+@pytest.mark.parametrize(
+    "get_credential",
+    [
+        auth.Resolver.get_username_from_keyring,
+        auth.Resolver.get_password_from_keyring,
+    ],
+)
+def test_log_exception_on_keyring_failure(get_credential, config, monkeypatch, caplog):
+    class FailKeyring:
+        @staticmethod
+        def get_credential(system, username):
+            # Simulate the error from https://github.com/pypa/twine/issues/889
+            environ = {}
+            environ["HOME"]
```

Comment:
This doesn't actually test `get_password_from_keyring`. It only causes failures for username; it raises an exception for `get_password_from_keyring` because that method tries to get the username from keyring if it's not already entered. Note the difference in fixtures in the previous two tests.